src/editors/Flow/view/Flow.jsx

```diff
@@ -89,14 +104,12 @@ export const Flow = (props, ref) => {
   );
 
   // State Hooks
-  const [loading, setLoading] = useState(true);
   const [dataFromDB, setDataFromDB] = useState();
   const [robotSelected, setRobotSelected] = useState("");
   const [runningFlow, setRunningFlow] = useState("");
   const [warnings, setWarnings] = useState([]);
   const [flowDebugging, setFlowDebugging] = useState();
-  const [warningsVisibility, setWarningsVisibility] = useState(false);
-  const [viewMode, setViewMode] = useState(FLOW_VIEW_MODE.default);
+  const viewMode = mainInterface?.viewMode ?? "default";
```

Comment:
This is no longer a state hook, so, on every render this `viewMode` variable will be recreated.

Reply:
Ok, we can use useMemo to avoid that.